fineract-provider/src/main/java/org/apache/fineract/infrastructure/bulkimport/importhandler/chartofaccounts/ChartOfAccountsImportHandler.java

```diff
@@ -0,0 +1,185 @@
+import org.apache.fineract.infrastructure.core.exception.*;
+import org.apache.poi.ss.usermodel.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+@Service
+public class ChartOfAccountsImportHandler implements ImportHandler {
+    private  List<GLAccountData> glAccounts;
+    private  Workbook workbook;
+
+    private final PortfolioCommandSourceWritePlatformService commandsSourceWritePlatformService;
+
+    @Autowired
+    public ChartOfAccountsImportHandler(final PortfolioCommandSourceWritePlatformService
+            commandsSourceWritePlatformService) {
+        this.commandsSourceWritePlatformService = commandsSourceWritePlatformService;
+    }
+
+    @Override
+    public Count process(Workbook workbook, String locale, String dateFormat) {
+        this.glAccounts=new ArrayList<>();
+        this.workbook=workbook;
+        readExcelFile();
+        return importEntity();
+    }
+
+    public void readExcelFile() {
+
+        Sheet chartOfAccountsSheet=workbook.getSheet(TemplatePopulateImportConstants.CHART_OF_ACCOUNTS_SHEET_NAME);
+        Integer noOfEntries= ImportHandlerUtils.getNumberOfRows(chartOfAccountsSheet,TemplatePopulateImportConstants.FIRST_COLUMN_INDEX);
+        for (int rowIndex=1;rowIndex<=noOfEntries;rowIndex++){
+            Row row;
+                row=chartOfAccountsSheet.getRow(rowIndex);
+                if (ImportHandlerUtils.isNotImported(row, ChartOfAcountsConstants.STATUS_COL)){
+                    glAccounts.add(readGlAccounts(row));
+                }
+        }
+    }
+
+    private GLAccountData readGlAccounts(Row row) {
+        String accountType=ImportHandlerUtils.readAsString(ChartOfAcountsConstants.ACCOUNT_TYPE_COL,row);
+        Long accountTypeId=null;
+        EnumOptionData accountTypeEnum=null;
+        if (accountType!=null && accountType.equalsIgnoreCase(GLAccountType.ASSET.toString())){
+            accountTypeId=1L;
+            accountTypeEnum=new EnumOptionData(accountTypeId,null,null);
+        }else if(accountType!=null && accountType.equalsIgnoreCase(GLAccountType.LIABILITY.toString())){
+            accountTypeId=2L;
+            accountTypeEnum=new EnumOptionData(accountTypeId,null,null);
+        }else if(accountType!=null && accountType.equalsIgnoreCase(GLAccountType.EQUITY.toString())){
+            accountTypeId=3L;
+            accountTypeEnum=new EnumOptionData(accountTypeId,null,null);
+        }
+        else if(accountType!=null && accountType.equalsIgnoreCase(GLAccountType.INCOME.toString())){
+            accountTypeId=4L;
+            accountTypeEnum=new EnumOptionData(accountTypeId,null,null);
+        }else if(accountType!=null && accountType.equalsIgnoreCase(GLAccountType.EXPENSE.toString())){
+            accountTypeId=5L;
+            accountTypeEnum=new EnumOptionData(accountTypeId,null,null);
+        }
```

Comment:
Move this code to GLAccountType new function fromString which will return the accountTypeEnum.